rust_src/src/keymap.rs

```diff
@@ -565,4 +568,42 @@ pub fn make_sparse_keymap(string: LispObject) -> LispObject {
     }
 }
 
+#[no_mangle]
+pub extern "C" fn describe_vector_princ(elt: LispObject, fun: LispObject) {
+    unsafe { Findent_to(LispObject::from_fixnum(16), LispObject::from_fixnum(1)) };
+    call!(fun, elt);
+    unsafe { Fterpri(Qnil, Qnil) };
+}
+
+/// Insert a description of contents of VECTOR.
+/// This is text showing the elements of vector matched against indices.
+/// DESCRIBER is the output function used; nil means use `princ'.
+#[lisp_fn(min = "1", name = "describe-vector")]
+pub fn describe_vector_lisp(vector: LispObject, mut describer: LispObject) {
```

Comment:
I recently introduced `map_or` and `map_or_else` on `LispObject`. However, neither really fits this situation. I am not sold on `unwrap_or_else` as the right name. Something like `obj.or_default(|| action())` maybe?